src/ObserverPattern/Subject/Account.cs

```diff
@@ -0,0 +1,37 @@
+﻿namespace ObserverPattern.Subject
+{
+    using System.Collections.Generic;
+    using Observer;
+
+    public abstract class Account : IAccount, IObservable<IBankClientObserver>
+    {
+        private readonly IList<IBankClientObserver> observers;
+
+        protected Account(string type)
+        {
+            this.Type = type;
+            this.observers = new List<IBankClientObserver>();
+        }
+
+        public string Type { get; }
+
+        public void Attach(IBankClientObserver observer)
+        {
+            this.observers.Add(observer);
+        }
+
+        public void Detach(IBankClientObserver
+            observer)
+        {
+            this.observers.Remove(observer);
+        }
+
+        public void Notify(string message)
```

Comment:
maybe this could be a protected method with a generic parameter Notify<T>(T event) which would call observer.Update(event) and the subclass BankAccount would be responsible for constructing the event object and calling this method?

Reply:
Yep protected virtual, in case another case inherits BankAccount. Yes the BankAccount class needs to implement the event in order to notify its observers. It holds the responsibility on how to notify.

Reply:
It must be protected for sure. I'll will also make it virtual and add the events.